src/components/SettingsPage/SettingsSyncCard.tsx

```diff
@@ -43,23 +52,140 @@ const SyncButton = ({ syncEntities }: SyncButtonProps) => {
   );
 };
 
+interface CancelSyncButtonProps {
+  cancelSync: () => Promise<void>;
+}
+
+const CancelSyncButton: React.FC<CancelSyncButtonProps> = ({ cancelSync }) => {
+  const [isCanceling, setIsCanceling] = useState(false);
+  return (
+    <IconButton
+      onClick={() => {
+        setIsCanceling(true);
+        cancelSync().finally(() => setIsCanceling(false));
+      }}
+      aria-busy={isCanceling}
+      aria-label="Cancel entity sync"
+    >
+      {isCanceling ? <CircularProgress /> : <CancelIcon />}
+    </IconButton>
+  );
+};
+
 export const SettingsSyncCard = () => {
   const catalogApi = useApi(catalogApiRef);
   const cortexApi = useApi(cortexApiRef);
   const extensionApi = useApi(extensionApiRef);
 
+  const [syncTaskProgressPercentage, setSyncTaskProgressPercentage] = useState<
+    number | null
+  >(null);
+
+  const [lastSyncedTime, setLastSyncedTime] = useState<string | null>(null);
+
   const syncEntities = async () => {
     const { items: entities } = await catalogApi.getEntities();
     const customMappings = await extensionApi.getCustomMappings?.();
     const groupOverrides = await extensionApi.getTeamOverrides?.(entities);
-    await cortexApi.syncEntities(entities, customMappings, groupOverrides);
+    const progress = await cortexApi.submitSyncTask(
+      entities,
+      customMappings,
+      groupOverrides,
+    );
+    setSyncTaskProgressPercentage(progress.percentage);
+  };
+
+  const cancelSync = async () => {
+    await cortexApi.cancelSync();
+    setSyncTaskProgressPercentage(null);
+    const lastSyncedTime = (await cortexApi.getLastSyncTime()).lastSynced;
+    setLastSyncedTime(lastSyncedTime);
+  };
+
+  const updateSyncProgressAndLastSyncedTime = async () => {
+    const currentProgress = await cortexApi.getSyncTaskProgress();
+    if (
+      syncTaskProgressPercentage !== null &&
+      currentProgress.percentage === null
+    ) {
+      setLastSyncedTime((await cortexApi.getLastSyncTime()).lastSynced);
+    }
+    setSyncTaskProgressPercentage(currentProgress.percentage);
   };
 
+  const updateSyncProgressAndLastSyncedTimeCallback = useCallback(
+    updateSyncProgressAndLastSyncedTime,
+    [cortexApi, syncTaskProgressPercentage],
+  );
```

Comment:
Why is this defined separately from the method itself? Why not just do
```suggestion
  const updateSyncProgressAndLastSyncedTime = useCallback(async () => {
    // implementation
  }, [cortexApi, syncTaskProgressPercentage]);
```

Reply:
That's because `updateSyncProgressAndLastSyncedTimeCallback` is being called in the `PollingLinearGauge` poll parameter.

Reply:
If we define `updaetSyncProgressAndLastSyncedTime` as a callback, then we can just call it directly, right?